### PR TITLE
Use GitHub Actions and not Travis CI for CI/CD

### PR DESCRIPTION
**Pull request by HenryFBP**
Switched to GitHub Actions for CI/CD.

Note that some tests:
- specific network tests fail on Windows, 
- a timing test fails on OSX, 
- but they both pass on Linux. 

Not sure if this is intended. See https://github.com/HenryFBP/repy_v2/actions/runs/1596790875 for logs.

Note that GH Actions doesn't natively support Python <2.7.17. Could probably manually add these.

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Run tests
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ['ubuntu-18.04', 'ubuntu-latest', 'macos-11', 'windows-2016']
```
Comment (lukpueh) [inline .github/workflows/tests.yaml:10]: Is there a reason you test on 2 ubuntu versions and on specific older macos and windows versions? Would the following also work?
```suggestion
        os: [ubuntu-latest, macos-latest, windows-latest]
```

File: .travis.yml [hunk 2]
```diff
@@ -63,7 +63,7 @@ matrix:
     #   docs.travis-ci.com/user/multi-os/#Python-example-%28unsupported-languages%29
     # OS X Python current version (currently 2.7.11)
     - language: generic
-      os: osx
+      os: osx
```
Comment (lukpueh) [inline .travis.yml:66]: IMO it's okay to just remove this file.

**Review by lukpueh — APPROVED** [review]
Thanks for the PR! LGTM. I've got two minor requests. Feel free to address, otherwise we can merge as is. Also, I don't think it's worth to test below 2.7.

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (HenryFBP) [reply .github/workflows/tests.yaml:10]: @lukpueh I just put random versions in to test compat 😄 you can change this to any valid OS

File: .travis.yml (hunk 2 above)
Comment (HenryFBP) [reply .travis.yml:66]: @lukpueh  Cool, will delete from branch.

**Comment by HenryFBP** [comment]
@lukpueh FYI, list of runners is here. https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#github-hosted-runners

`rm .travis.yml` and also using `        os: [ubuntu-latest, macos-latest, windows-latest]` now.

**Comment by lukpueh** [comment]
Thanks for the updates, @HenryFBP! Given that `macos-latest` (understandably!) doesn't ship Python2 anymore, I suggest to just revert to `macos-11`. You could also update the workflow to install Python2 on macos-latest. But I wouldn't spend too much effort on keeping the long-dead snake alive.

**Comment by HenryFBP** [comment]
> Thanks for the updates, @HenryFBP! Given that `macos-latest` (understandably!) doesn't ship Python2 anymore, I suggest to just revert to `macos-11`. You could also update the workflow to install Python2 on macos-latest. But I wouldn't spend too much effort on keeping the long-dead snake alive.

No prob. Changed back to `macos-11` from `macos-latest`.

**Pull request merged**